### PR TITLE
enh: getAgentConfgurations use joins to fetch dataSources and Workspace

### PR DESCRIPTION
**Pull request by fontanierh**
Adding the joins back saves 2 DB roundtrips, and the joins are always done on a small amount of agents configurations.

### CODE REVIEW

File: front/lib/api/assistant/configuration.ts [hunk 1]
```diff
@@ -283,6 +283,18 @@ export async function getAgentConfigurations<V extends "light" | "full">({
               [Op.in]: Object.values(retrievalConfigs).map((r) => r.id),
             },
           },
+          include: [
```
Comment (spolu) [inline front/lib/api/assistant/configuration.ts:286]: LGTM but aren't we supposed to run that only below line `if (variant === full)` ?

**Review by spolu — APPROVED** [review]
LGTM

File: front/lib/api/assistant/configuration.ts (hunk 1 above)
Comment (fontanierh) [reply front/lib/api/assistant/configuration.ts:286]: If you expand the diff to see the beginning of the statement, you'll see that we only do this if `retrievalConfigs` is non-empty (which only happens if we're in the full variant case)

Comment (spolu) [reply front/lib/api/assistant/configuration.ts:286]: Yep kinda hard to follow as commented on the other PR as well. But trusting you.

Comment (fontanierh) [reply front/lib/api/assistant/configuration.ts:286]: IMHO when you have the full code (and not this small diff) in front of you it feels pretty obvious. I can add a comment if you feel it's useful

Comment (spolu) [reply front/lib/api/assistant/configuration.ts:286]: I was looking at it with the full code. Thing that didn't seem obvious is why that code is not below the "if full".

Comment (fontanierh) [reply front/lib/api/assistant/configuration.ts:286]: That would indent everything a little bit more to the right for now reason, given that we still need to check if there are any retrievalConfigs before fetching anyway

Comment (fontanierh) [reply front/lib/api/assistant/configuration.ts:286]: Will add comments then 👍
